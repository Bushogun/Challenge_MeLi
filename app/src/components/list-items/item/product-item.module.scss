@use '../../../styles/colors.module.scss' as *;

.product_container{
    margin-top: 4em; 
    margin-bottom: 10px;
    hr{
        width: 95%;
        border-color: #fff;
    }
    .info_container{ 
        display: flex;

        .container_sell_info{
            img{
                margin: 15px;
            }
            .container_state_info{
                display: flex;
                justify-content: center;
                align-items: center;
                .free_shipping{
                    padding-top: 5px;
                    color: $color__green--1;
                    font-size: 30px;
                    cursor: pointer;
                }
                .condition{
                    padding: 5px;
                    background-color: $color__green--1;
                    color: #fff;
                    margin: 0px 10px ;
                    border-radius: 5px;
                    cursor: pointer;

                }
                .used{
                    background-color: #2236bb;
                }
            }
        }
    }

    .product_description{
        display: block; 
        .title_price{
            padding-top: 3rem;
            font-size: 2.6em;
            color: $color__gray--1;
        }
        .title{
            padding-top: 10px;
            font-size: 1.6em;
            max-width: 20em;
            color: $color__gray--2;
        }
        .quantity{
            padding-top: 10px;
            font-size: 1.1em;
            font-weight: 500;
            color: $color__green--1;
        }
    }

    .container_details{
        display: block;
        margin-left: auto;
        margin-right: auto;
        min-width: 10em;
        .state_name {
            padding-top: 3rem;
            font-size: 1em;
            color: $color__gray--3;
        }
        .city_name {
            font-size: 1em;
            color: $color__gray--3;
        }
    }

    @media screen and (max-width: 768px) {
        hr{
            width: 88%;
        }
        .info_container{ 
            display: block;
            .container_sell_info{
                img{
                    display: flex;
                    margin: auto;
                    margin-top: 2em;
                }
                display: block;
                align-items: center;
                .container_state_info{
                    display: flex;
                }
            }
	    }

        .product_description{
            padding: 0em 5em;
            .title_price{
                text-align: center;
                padding-top: 1rem;
                font-size: 2.2em;
            }
            .title{
                text-align: center;
                padding-top: 8px;
                font-size: 1.2em;
            }
            .quantity{
                text-align: center;
                padding-top: 5px;
                font-size: 0.9em;
                color: $color__green--1;
            }
        }

        .container_details{
            display: flex;
            justify-content: space-around;
            padding-top: 15px;
            .state_name {
                padding-top: 0rem;
                font-size: 1em;
            }
            .city_name {
                font-size: 1em;
            }
        }

    }

    @media screen and (max-width: 422px) {
        hr{
            width: 80%;
        }
	
        .info_container{ 
            display: block;
            .container_sell_info{
                img{
                    display: flex;
                    margin: auto;
                    margin-top: 2em;
                }
                display: block;
                align-items: center;
                .container_state_info{
                    display: flex;
                }
            }
        }

        .product_description{
            padding: 0em 1em;
            .title_price{
                text-align: center;
                padding-top: 1rem;
                font-size: 1.8em;
            }
            .title{
                text-align: center;
                padding-top: 6px;
                font-size: 1em;
            }
            .quantity{
                text-align: center;
                padding-top: 6px;
                font-size: 0.9em;
                color: $color__green--1;
            }
        }

        .container_details{
            display: flex;
            justify-content: space-around;
            padding-top: 10px;
            .state_name {
                padding-top: 0rem;
                font-size: 0.8em;
            }
            .city_name {
                font-size: 0.8em;
            }
        }
    }

	@media screen and (max-width: 320px) {
        hr{
            width: 80%;
        }
	
        .info_container{ 
            display: block;
            .container_sell_info{
                img{
                    display: flex;
                    margin: auto;
                    margin-top: 2em;
                }
                display: block;
                align-items: center;
                .container_state_info{
                    display: flex;
                }
            }
        }

        .product_description{
            padding: 0em 1em;
            .title_price{
                text-align: center;
                padding-top: 1rem;
                font-size: 1.8em;
            }
            .title{
                text-align: center;
                padding-top: 6px;
                font-size: 1em;
            }
            .quantity{
                text-align: center;
                padding-top: 6px;
                font-size: 0.9em;
                color: $color__green--1;
            }
        }

        .container_details{
            display: flex;
            justify-content: space-around;
            padding-top: 10px;
            .state_name {
                padding-top: 0rem;
                font-size: 0.8em;
            }
            .city_name {
                font-size: 0.8em;
            }
        }
    }
}