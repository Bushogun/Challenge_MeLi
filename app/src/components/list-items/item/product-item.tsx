import React from 'react';
import { TbTruckDelivery } from 'react-icons/tb'
import styles from './product-item.module.scss'


export const ProductItem = () => {
    return (
      <div className={styles.product_container}>
        <div className={styles.info_container}>
          <div className={styles.container_sell_info}>
            <img src="https://http2.mlstatic.com/D_619667-MLA47781882790_102021-I.jpg"
            alt="image product  Apple iPhone 13 (128 Gb) - Azul" width="180" height="180" />
            <div className={styles.container_state_info}>
              <p className={styles.condition} title="Atículo nuevo">New</p>
              <p className={styles.free_shipping} title="Envío gratis"><TbTruckDelivery /></p> 
            </div>
          </div>
          <div className={styles.product_description}>
            <h1 className={styles.title_price}>$754000</h1>
            <h2 className={styles.title}>Apple iPhone 13 (128 Gb) - Azul</h2>
            <p className={styles.quantity}> En 12 cuotas de $141752</p>
            {/* <p className={styles.amount}>Cuotas desde: $ 141752</p> */}
          </div>  

          <div className={styles.container_details}>
            <p className={styles.state_name}>Capital Federal</p>
            <p className={styles.city_name}>Colegiales</p>
          </div>
          
        </div>
          <hr />
      </div>
    );
  };

