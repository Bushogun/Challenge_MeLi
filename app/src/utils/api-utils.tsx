import { setLoading } from '@/src/store/productSlice';

export async function fetchProductData(
  dispatch: any,
  searchQuery: string,
  limit: string,
  selectedSort: string,
  priceFilter: string,
) {
  try {
    dispatch(setLoading(true));

    let sortQueryParam = '';
    let priceQueryParam = '';
    if (selectedSort) {
      sortQueryParam = `&sort=${selectedSort}`;
    }
    if (priceFilter) {
      priceQueryParam = `&price=${priceFilter}`;
    }

    const response = await fetch(
      `https://api.mercadolibre.com/sites/MLA/search?q=${searchQuery}&limit=${limit}${sortQueryParam}${priceQueryParam}`
    );
    console.log (response);
    if (!response.ok) {
      throw new Error('Error al cargar productos');
    }

    const data = await response.json();
    dispatch(setLoading(false));

    return data;
  } catch (error) {
    dispatch(setLoading(false));
    throw error;
  }
}
