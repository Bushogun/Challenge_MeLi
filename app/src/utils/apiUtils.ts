
export async function fetchProductData(
    searchQuery: string,
    limit: number,
    selectedSort: string | null,
    priceFilter: string | null
  ) {
    const sortQueryParam = selectedSort ? `&sort=${selectedSort}` : '';
    const priceQueryParam = priceFilter ? `&price=${priceFilter}` : '';
  
    const response = await fetch(
      `https://api.mercadolibre.com/sites/MLA/search?q=${searchQuery}&limit=${limit}${sortQueryParam}${priceQueryParam}`
    );
  
    if (!response.ok) {
      throw new Error('Error al cargar productos');
    }
  
    const data = await response.json();
    return data;
  }

/*
export async function fetchProducts(
    searchQuery: string,
    limit: number,
    selectedSort: string | null,
    priceFilter: string | null
  ) {
    const sortQueryParam = selectedSort ? `&sort=${selectedSort}` : '';
    const priceQueryParam = priceFilter ? `&price=${priceFilter}` : '';
  
    const response = await fetch(
      `https://api.mercadolibre.com/sites/MLA/search?q=${searchQuery}&limit=${limit}${sortQueryParam}${priceQueryParam}`
    );
  
    if (!response.ok) {
      throw new Error('Error al cargar productos');
    }
  
    const data = await response.json();
    return data.results;
  }
  
  export async function fetchAvailableSorts(
    searchQuery: string,
    limit: number,
    selectedSort: string | null
  ) {
    const sortQueryParam = selectedSort ? `&sort=${selectedSort}` : '';
  
    const response = await fetch(
      `https://api.mercadolibre.com/sites/MLA/search?q=${searchQuery}&limit=${limit}${sortQueryParam}`
    );
  
    if (!response.ok) {
      throw new Error('Error al cargar available_sorts');
    }
  
    const data = await response.json();
    return data.available_sorts;
  }
  
  export async function fetchAvailablePriceFilter(searchQuery: string, limit: number) {
    const response = await fetch(
      `https://api.mercadolibre.com/sites/MLA/search?q=${searchQuery}&limit=${limit}`
    );
  
    if (!response.ok) {
      throw new Error('Error al cargar available_filters');
    }
  
    const data = await response.json();
    return data.available_filters.find((filter: any) => filter.id === 'price');
  }
  */