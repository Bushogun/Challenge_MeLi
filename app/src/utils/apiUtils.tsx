import { useDispatch } from 'react-redux';
import { setProducts, setLoading, setError } from '@/src/store/productSlice';

export async function fetchProductData(
  dispatch: any,
  searchQuery: string,
  limit: string,
  selectedSort: string,
  priceFilter: string,
) {
  try {
    dispatch(setLoading(true));

    let sortQueryParam = '';
    let priceQueryParam = '';
    if (selectedSort) {
      sortQueryParam = `&sort=${selectedSort}`;
    }
    if (priceFilter) {
      priceQueryParam = `&price=${priceFilter}`;
    }

    const response = await fetch(
      `https://api.mercadolibre.com/sites/MLA/search?q=${searchQuery}&limit=${limit}${sortQueryParam}${priceQueryParam}`
    );
    if (!response.ok) {
      throw new Error('Error al cargar productos');
    }

    const data = await response.json();
    dispatch(setLoading(false));

    return data;
  } catch (error) {
    dispatch(setLoading(false));
    throw error;
  }
}
