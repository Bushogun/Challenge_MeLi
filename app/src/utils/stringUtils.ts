export function capitalizeLocations(text: string): string {
    return text.replace(/\b[A-Z]+\b/g, (word) => {
        return word.charAt(0) + word.slice(1).toLowerCase();
    });
}

export function capitalizeTitle(text: string): string {
    return text
        .toLowerCase()
        .split(' ')
        .map((word) => word.charAt(0).toUpperCase() + word.slice(1))
        .join(' ');
}
