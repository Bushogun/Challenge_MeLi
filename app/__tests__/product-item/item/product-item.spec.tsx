import React from 'react';
import { renderWithProviders } from '@/src/utils/test-utils';
import { ProductItem } from '@/src/components/list-items/item/product-item';
import { productMock } from '@/__mocks__/product-detail.mock';

describe('ProductItem component', () => {
  it('should render correctly', () => {
    const { container } =  renderWithProviders(<ProductItem product={productMock} />);
    expect(container).toMatchSnapshot();
  });
});


