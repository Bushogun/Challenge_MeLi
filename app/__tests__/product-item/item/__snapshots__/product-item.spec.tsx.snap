// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductItem component should render correctly 1`] = `
<div>
  <div
    class="product_container"
  >
    <div
      class="info_container"
    >
      <div
        class="container_sell_info"
      >
        <img
          alt="Imagen Apple iPhone 11 (128 Gb) - Negro"
          height="180"
          src="http://http2.mlstatic.com/D_865864-MLA46114990464_052021-I.jpg"
          width="180"
        />
        <div
          class="container_state_info"
        >
          <p
            class="condition "
            title="Artículo nuevo"
          >
            new
          </p>
          <p
            class="free_shipping"
            title="Envío gratis"
          >
            <svg
              fill="none"
              height="1em"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 0h24v24H0z"
                fill="none"
                stroke="none"
              />
              <path
                d="M7 17m-2 0a2 2 0 1 0 4 0a2 2 0 1 0 -4 0"
              />
              <path
                d="M17 17m-2 0a2 2 0 1 0 4 0a2 2 0 1 0 -4 0"
              />
              <path
                d="M5 17h-2v-4m-1 -8h11v12m-4 0h6m4 0h2v-6h-8m0 -5h5l3 5"
              />
              <path
                d="M3 9l4 0"
              />
            </svg>
          </p>
        </div>
      </div>
      <div
        class="product_description"
      >
        <h1
          class="title_price"
        >
          $ 
          501.484
        </h1>
        <h2
          class="title"
        >
          Apple Iphone 11 (128 Gb) - Negro
        </h2>
        <p
          class="quantity"
        >
          En 
          12
           cuotas de $ 
          94278.99
        </p>
      </div>
      <div
        class="container_details"
      >
        <p
          class="state_name"
        >
          Capital Federal
        </p>
        <p
          class="city_name"
        >
          Capital Federal
        </p>
      </div>
    </div>
    <hr />
  </div>
</div>
`;
